// naver-login/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Cordova, IonicNativePlugin, Plugin } from '@ionic-native/core';
import { Injectable } from '@angular/core';
var NaverLogin = (function (_super) {
    __extends(NaverLogin, _super);
    function NaverLogin() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NaverLogin.prototype.login = function () {
        return;
    };
    NaverLogin.prototype.logout = function () {
        return;
    };
    NaverLogin.prototype.unlinkApp = function () {
        return;
    };
    NaverLogin.prototype.refreshAccessToken = function () {
        return;
    };
    NaverLogin.prototype.getAccessToken = function () {
        return;
    };
    NaverLogin.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    NaverLogin.ctorParameters = function () { return []; };
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], NaverLogin.prototype, "login", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], NaverLogin.prototype, "logout", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], NaverLogin.prototype, "unlinkApp", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], NaverLogin.prototype, "refreshAccessToken", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], NaverLogin.prototype, "getAccessToken", null);
    NaverLogin = __decorate([
        Plugin({
            pluginName: 'NaverLogin',
            plugin: 'cordova-plugin-naverlogin',
            pluginRef: 'NaverLogin',
            platforms: ['Android', 'ios']
        })
    ], NaverLogin);
    return NaverLogin;
}(IonicNativePlugin));
export { NaverLogin };
